### ISSUE TITLE
Style/MutableConstant false positive

### ISSUE THREAD

**Issue by davidhrbac** [rubocop-hq/rubocop#6135]
## Expected behavior

List is being augmented with `default` after declaration.

## Actual behavior

```console
jobs/google_spreadsheet.rb:15:15: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
SUPERVISORS = { ...
```

## Steps to reproduce the problem

```ruby
SUPERVISORS = {
  'HRB' => 'David Hrbáč',
  'DOE' => 'John Doe',
}
SUPERVISORS.default = 'N/A'
```

## RuboCop version

```console
$ rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.4.3 x86_64-linux)
```

**Comment by asherkach** [comment]
This is "working as intended". The cop infers that `SUPERVISORS` is a constant from the fact that it is in [SCREAMING_CASE](https://github.com/rubocop-hq/ruby-style-guide#screaming-snake-case), so wants the object to be frozen to prevent mutation after declaration (as you are doing by setting the default on a later line).

Since `SUPERVISORS` is not a constant, perhaps we should be flagging the line for [naming convention](https://github.com/rubocop-hq/ruby-style-guide#snake-case-symbols-methods-vars)?

**Comment by davidhrbac** [comment]
@asherkach yes, you are right... my fault, closing the ticket.
